erpnext/stock/report/item_prices/item_prices.py

```diff
@@ -82,33 +82,34 @@ def get_last_purchase_rate():
 
 	query = """select * from (select
 					result.item_code,
-					result.base_rate
+					result.base_net_rate
 					from (
 						(select
 							po_item.item_code,
-							po_item.item_name,
-							po.transaction_date as posting_date,
-							po_item.base_price_list_rate,
-							po_item.discount_percentage,
-							po_item.base_rate
+							po_item.base_net_rate
 						from `tabPurchase Order` po, `tabPurchase Order Item` po_item
-						where po.name = po_item.parent and po.docstatus = 1)
+						where po.name = po_item.parent and po.docstatus = 1
+						order by po.transaction_date desc, po.name desc limit 1)
 						union
 						(select
 							pr_item.item_code,
-							pr_item.item_name,
-							pr.posting_date,
-							pr_item.base_price_list_rate,
-							pr_item.discount_percentage,
-							pr_item.base_rate
+							pr_item.base_net_rate
 						from `tabPurchase Receipt` pr, `tabPurchase Receipt Item` pr_item
-						where pr.name = pr_item.parent and pr.docstatus = 1)
+						where pr.name = pr_item.parent and pr.docstatus = 1
+						order by pr.posting_date desc, pr.posting_time desc, pr.name desc limit 1)
```

Comment:
Same as above - this `limit 1` is going to dump only a single row from the output - then we will never be creating the *all item map* which is then consumed by calling function to dump the report.
I might be missing something really big here which is why I am wondering - did you get the report correct after this?